packages/ua-devtools-evm/src/oapp/sdk.ts

```diff
@@ -71,42 +71,49 @@ export class OApp extends OmniSDK implements IOApp {
         this.logger.debug(`Getting enforced options for eid ${eid} (${formatEid(eid)}) and message type ${msgType}`)
 
         return await this.contract.contract.enforcedOptions(eid, msgType)
     }
 
-    async setEnforcedOptions(enforcedOptions: EnforcedOptions[]): Promise<OmniTransaction> {
+    async setEnforcedOptions(enforcedOptions: OAppEnforcedOptionParam[]): Promise<OmniTransaction> {
         this.logger.debug(`Setting enforced options to ${printJson(enforcedOptions)}`)
+        const serializedConfig = this.serializeExecutorOptions(enforcedOptions)
 
-        const data = this.contract.contract.interface.encodeFunctionData('setEnforcedOptions', [enforcedOptions])
+        const data = this.contract.contract.interface.encodeFunctionData('setEnforcedOptions', [serializedConfig])
         return {
             ...this.createTransaction(data),
             description: `Setting enforced options to ${printJson(enforcedOptions)}`,
         }
     }
 
-    encodeEnforcedOptions(enforcedOptionConfig: OAppEnforcedOptionConfig): Options {
-        if ('options' in enforcedOptionConfig) return Options.fromOptions(enforcedOptionConfig.options)
-
-        if (enforcedOptionConfig.msgType == ExecutorOptionType.LZ_RECEIVE) {
-            return Options.newOptions().addExecutorLzReceiveOption(enforcedOptionConfig.gas, enforcedOptionConfig.value)
-        } else if (enforcedOptionConfig.msgType == ExecutorOptionType.NATIVE_DROP) {
-            return Options.newOptions().addExecutorNativeDropOption(
-                enforcedOptionConfig.amount,
-                enforcedOptionConfig.receiver
-            )
-        } else if (enforcedOptionConfig.msgType == ExecutorOptionType.COMPOSE) {
-            return Options.newOptions().addExecutorComposeOption(
-                enforcedOptionConfig.index,
-                enforcedOptionConfig.gas,
-                enforcedOptionConfig.value
-            )
-        } else if (enforcedOptionConfig.msgType == ExecutorOptionType.ORDERED) {
-            return Options.newOptions().addExecutorOrderedExecutionOption()
-        } else {
-            throw new Error(`Invalid ExecutorOptionType`)
+    /**
+     * Prepares the Executor config to be sent to the contract
+     *
+     * @param {OAppEnforcedOptionParam[]}
+     * @returns {SerializedEnforcedOptions[]}
+     */
+    protected serializeExecutorOptions(
+        oappEnforcedOptionParam: OAppEnforcedOptionParam[]
+    ): SerializedEnforcedOptions[] {
+        const serializedEnforcedOptions: SerializedEnforcedOptions[] = []
+        for (const oAppEnforcedOptionParam of oappEnforcedOptionParam) {
+            serializedEnforcedOptions.push({
+                eid: oAppEnforcedOptionParam.eid,
+                msgType: oAppEnforcedOptionParam.option.msgType,
+                options: oAppEnforcedOptionParam.option.options,
+            })
         }
+        return serializedEnforcedOptions
```

Comment:
This is just a code-style thing - in a situation like this, I'd normally go for a `.map` rather than pushing into an array. There is nothing wrong with this approach ofc and it works as expected - what `.map` enforces is a concept of immutability - basically the fact that once an object (an array in this case) is created, it is not being changed. The code also gets a bit smaller:

```typescript
return oappEnforcedOptionParam.map(({ eid, option: { msgType, options } }) => ({ eid, msgType, options })
```